kyo-prelude/shared/src/main/scala/kyo/Parse.scala

```diff
@@ -36,18 +36,24 @@ import kyo.kernel.*
   * @see
   *   [[kyo.Var]], [[kyo.Choice]], [[kyo.Abort]] for the underlying effects that Parse composes
   */
-opaque type Parse <: (Var[Parse.State] & Choice & Abort[ParseFailed]) = Var[Parse.State] & Choice & Abort[ParseFailed]
+opaque type Parse[A] <: (Var[Parse.State[A]] & Choice & Abort[ParseFailed]) =
+    Var[Parse.State[A]] & Choice & Abort[ParseFailed]
```

Comment:
Thanks a lot for this PR. Maybe there is a trick with two type param, `Parse[-A1, +A2]`, but I am not sure. 

@johnhungerford What do you think?